src/foam/dao/F3FileJournal.js

```diff
@@ -39,53 +41,63 @@ foam.CLASS({
       ],
       javaCode: `
         // count number of entries successfully read
-        int successReading = 0;
-        JSONParser parser = getParser();
+        AtomicInteger successReading = new AtomicInteger();
```

Comment:
This can just be a regular 'int'. It is only updated in endJob() which is single-threaded so doesn't need synchronization.